src/amo/withExperiment.js

```diff
@@ -215,6 +282,9 @@ export const withExperiment =
         if (cleanupNeeded || addExperimentToCookie) {
           cookies.set(EXPERIMENT_COOKIE_NAME, experimentsToStore, cookieConfig);
         }
+
+        // Remove the stored variant from the Redux store.
+        dispatch(storeExperimentVariant({ storedVariant: null }));
```

Comment:
I thought about including a `clearExperimentVariant` action for this, but that also seemed like overkill for something that will only ever be called from here, so I am reusing `storeExperimentVariant`. I can change that if you think it would be better.